File: Models/CvReference.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace MtpApp.Models
{
    public class CvReference
    {
        public int Id { get; set; }
        [Required]
        [StringLength(255)]
        public string Name { get; set; }
        public string Description { get; set; }

        [Required]
        public int ClientId { get; set; }
        public Client Client { get; set; }

        public int JobPositionId { get; set; }

        public JobPosition JobPositions { get; set; }
        public string Organization { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }

    }
}

File: spring-backend/src/main/java/com/mtp/api/models/CvReference.java
package com.mtp.api.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Entity
@Table(name = "CvReferences")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CvReference {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull
    @Size(max = 255)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ClientId")
    private Client client;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "JobPositionId")
    private JobPosition jobPosition;

    private String organization;
    private String phone;
    private String email;
}
